tests/compiler/test_warnings.py

```diff
@@ -48,11 +48,10 @@ def test_warnings(option: Optional[str], expected_error: bool, expected_warning:
         else:
             inmanta_warnings.warn(internal_warning)
         if expected_warning:
-            assert len(w) == 1
-            assert issubclass(w[0].category, CompilerRuntimeWarning)
-            assert str(w[0].message) == message
+            assert len(caught_warnings) >= 1
+            assert any([issubclass(w.category, CompilerRuntimeWarning) and str(w.message) == message for w in caught_warnings])
         else:
-            assert len(w) == 0
+            assert len(caught_warnings) == 0
```

Comment:
This constraint is still too strict. Here we should just check that the specific warning we are testing for is not present in `caught_warning`. But other warnings are allowed.